enterprise/backup/src/test/java/org/neo4j/backup/OnlineBackupCommandHaIT.java

```diff
@@ -97,27 +124,65 @@ public void makeSureBackupCanBePerformedWithCustomPort() throws Exception
         startDb( backupPort );
         assertEquals( "should not be able to do backup when noone is listening",
                 1,
-                runBackupTool( "--from", "127.0.0.1:" + PortAuthority.allocatePort(),
+                runBackupToolInSameJvm( "--from", "127.0.0.1:" + PortAuthority.allocatePort(),
                         "--cc-report-dir=" + backupDir,
                         "--backup-dir=" + backupDir,
                         "--name=" + backupName ) );
         assertEquals(
                 0,
-                runBackupTool( "--from", "127.0.0.1:" + backupPort,
+                runBackupToolInSameJvm( "--from", "127.0.0.1:" + backupPort,
                         "--cc-report-dir=" + backupDir,
                         "--backup-dir=" + backupDir,
                         "--name=" + backupName ) );
         assertEquals( getDbRepresentation(), getBackupDbRepresentation( backupName ) );
         createSomeData( db );
         assertEquals(
                 0,
-                runBackupTool( "--from", "127.0.0.1:" + backupPort,
+                runBackupToolInSameJvm( "--from", "127.0.0.1:" + backupPort,
                         "--cc-report-dir=" + backupDir,
                         "--backup-dir=" + backupDir,
                         "--name=" + backupName ) );
         assertEquals( getDbRepresentation(), getBackupDbRepresentation( backupName ) );
     }
 
+    @Test
+    public void dataIsInAUsableStateAfterBackup() throws Exception
+    {
+        // given database exists
+        int backupPort = PortAuthority.allocatePort();
+        startDb( backupPort );
+
+        // and the database has indexes
+        createIndexes( db );
+
+        // and the database is being populated
+        AtomicBoolean continueFlagReference = new AtomicBoolean( true );
+        new Thread( () -> repeatedlyPopulateDatabase( db, continueFlagReference ) ).start(); // populate db with number properties etc.
+        oneOffShutdownTasks.add( () -> continueFlagReference.set( false ) ); // kill thread
+
+        // then backup is successful
+        String ip = ":" + backupPort;
+        assertEquals( 0,
+                runBackupTool( "--from", ip, "--cc-report-dir=" + backupDir, "--backup-dir=" + backupDir, "--name=defaultport" ) );
+    }
+
+    private void repeatedlyPopulateDatabase( GraphDatabaseService db, AtomicBoolean continueFlagReference )
+    {
+        while ( continueFlagReference.get() )
+        {
+            createSomeData( db );
+        }
+    }
+
+    private void createIndexes( GraphDatabaseService db )
+    {
+        try ( Transaction tx = db.beginTx() )
```

Comment:
This also needs coreTx(..).

Reply:
This is in the HA part of the integration tests, therefore it doesn't have `coreTx`; Not addressed